Add Travis CI support

jsiloto opened the pull request.

Commit:
Add Travis CI support

mmagr commented on .travis.yml:
```diff
@@ -0,0 +1,21 @@
+- docker pull minio/minio
+- docker-compose -f local/compose.yml -p imgm up -d
+- docker build -f Dockerfile -t local/imagemanager .
+- docker run -d --rm -it --network imgm_default -p 8000:5000 --network-alias image-manager -v $PWD:/usr/src/app local/imagemanager
```
this is just so we can use the service for tests, right?
If so, I think we can drop a couple of options - should make things easier to follow:
> docker run -d --network imgm_default -p 8000:5000 local/imagemanager

---
- `--rm` - has no effect (travis will tear the whole thing down anyways
- `-it` - makes no sense: this is used if we want to attach the current terminal into the container (also redirecting stdin/stdout). Also, is overriden by `-d`
- `-v $PWD` we should make sure we're testing *exactly* what's versioned, so why using the custom mount? ("official" deployments won't.)
- `--network-alias` - since this is not used by any other service of the minimal test deployment, and we're running tests from the host, this is not needed (we'll use localhost:8000 in test.py anyways).

Commit:
Remove unnecessary docker flags

Commit:
Fix problematic versioning of Marshmallow

requirements accepted versions >=3.0.0b3 but API changed on 3.0.0b7, see:
http://marshmallow.readthedocs.io/en/latest/api_reference.html#marshmallow.Schema.load

giovannicuriel reviewed (commented):
Shouldn't tests/client.py return something different than 0 so that Travis would know if something went wrong?

jsiloto commented:
Yes, there should be a test outputting errors, I intend to send another PR with a proper test suite. So far this should check if the build is working and the client can send requests.

jsiloto committed:
Merge branch 'master' into master
